src/scripting/Elsa.Scripting.JavaScript/Converters/TruncatingNumberJsonConverter.cs

```diff
@@ -27,7 +23,7 @@ private static bool IsWholeValue(object value)
 
             if (value is float || value is double)
             {
-                var doubleValue = (double) value;
```

Comment:
Note that
```
object value = (float) 42;
var number = (double) value;
```

will throw an InvalidCastException, but Convert.ToDouble will not

Reply:
Good point.